webhooks/cluster.go

```diff
@@ -246,14 +254,20 @@ func (webhook *Cluster) validateTopology(ctx context.Context, old, new *clusterv
 				),
 			)
 		}
-	}
-	// Check to see if the ClusterClass referenced in the Cluster currently exists.
-	if err := webhook.Client.Get(ctx, client.ObjectKey{Namespace: new.Namespace, Name: new.Spec.Topology.Class}, &clusterv1.ClusterClass{}); err != nil {
-		allErrs = append(
-			allErrs, field.Invalid(
-				field.NewPath("spec", "topology", "class"),
-				new.Spec.Topology.Class,
-				"ClusterClass could not be found"))
+
+		// If the ClusterClass name has changed (ClusterClass rebase) additional compatibility checks are needed.
+		if old.Spec.Topology.Class != new.Spec.Topology.Class {
+			oldClusterClass := &clusterv1.ClusterClass{}
+			// Check to see if the ClusterClass referenced in the Cluster currently exists.
+			if err := webhook.Client.Get(ctx, client.ObjectKey{Namespace: old.Namespace, Name: old.Spec.Topology.Class}, oldClusterClass); err != nil {
+				allErrs = append(
+					allErrs, field.Forbidden(
+						field.NewPath("spec", "topology", "class"),
+						fmt.Sprintf("old ClusterClass %v could not be found when rebasing to %v", old.Name, new.Name)))
```

Comment:
```suggestion
						fmt.Sprintf("old ClusterClass %v could not be found, this makes impossible to validate changes to %v", old.Name, new.Name)))
```
trying to avoid rebase as a term